Mods/ArchipelagoPlayer.SC2Mod/Base.SC2Data/GameData/ActorData.xml

```diff
@@ -34233,10 +34256,15 @@
         <On Terms="Behavior.CarrierPurifierChargeBeamPhase2.On; IsStatus WeaponAttacking 1" Send="Create AP_CarrierPurifierLoopSound2"/>
         <On Terms="Behavior.CarrierPurifierChargeBeamPhase3.On" Target="AP_CarrierPurifierLoopSound2" Send="Destroy"/>
         <On Terms="Behavior.CarrierPurifierChargeBeamPhase3.On; IsStatus WeaponAttacking 1" Send="Create AP_CarrierPurifierLoopSound3"/>
+        <On Terms="Behavior.WarpInEffect.On" Send="ModelSwap AP_CarrierPurifierWarpInStargate"/>
+        <On Terms="Behavior.WarpInEffect.On" Send="$Spell"/>
+        <On Terms="AnimDone; AnimName Spell" Send="ModelSwap Carrier_Purifier"/>
+        <On Terms="UnitConstruction.AP_CarrierPurifier.Start" Send="$Stand 0 -1.000000 -1.000000 3.250000 AsDuration"/>
+        <On Terms="AnimDone; AnimName Stand" Send="$Spell 0 -1.000000 -1.000000 1.750000 AsDuration"/>

```

Comment:
Purifier Carrier still looked a little off when warping in. I think there's a signal for changing the animation scale, do you think you could use that to drag out the animation?

Another option is to try to overlay the stargate warp in animation with a normal model warp in. Would involve creating additional temporary actors which may be a little much for a small detail like this.